backend/controller/internal/dal/events.go

```diff
@@ -180,7 +180,7 @@ type eventRow struct {
 	RequestKey     types.Option[model.IngressRequestKey]
 }
 
-func (d *DAL) QueryEvents(ctx context.Context, filters ...EventFilter) ([]Event, error) {
+func (d *DAL) QueryEvents(ctx context.Context, limit *int, filters ...EventFilter) ([]Event, error) {
```

Comment:
Actually, if this truly is optional, it should just be a filter function like all the others.

Reply:
ok maybe I'll just go that route then. I wasn't sure if there was a common pattern here I should follow but it sounds like Filters might be the way to go